src/ds++/path_getFilenameComponent.cc

```diff
@@ -116,10 +116,11 @@ std::string getFilenameComponent(std::string const &fqName,
   }
 
   // Always convert paths to use native format.
-  if (dirSep == WinDirSep) // Windows style.
-    std::replace(retVal.begin(), retVal.end(), UnixDirSep, dirSep);
-  else
-    std::replace(retVal.begin(), retVal.end(), WinDirSep, dirSep);
+#ifdef WIN32
+  std::replace(retVal.begin(), retVal.end(), UnixDirSep, dirSep);
+#else
+  std::replace(retVal.begin(), retVal.end(), WinDirSep, dirSep);
```

Comment:
This conditional looks swapped to me - I would think it should be:
```
#ifdef WIN32
  std::replace(retVal.begin(), retVal.end(), WinDirSep, dirSep);
#else
  std::replace(retVal.begin(), retVal.end(), UnixDirSep, dirSep);
```
Apologies if I'm missing something.

Reply:
Oh - apologies, I was confusing the old and new string in the `std::replace` function.